src/tools/wpa/DataModel/QuicEventPayload.cs

```diff
@@ -279,4 +304,36 @@ public class QuicStreamOutFlowBlockedPayload
     }
 
     #endregion
+
+    #region Datapath Event Payloads
+
+    public class QuicDatapathSendPayload
+    {
+        public uint TotalSize { get; protected set; }
+
+        public byte BufferCount { get; protected set; }
+
+        public ushort SegmentSize { get; protected set; }
+
+#pragma warning disable CS8618 // Non-nullable field is uninitialized.
```

Comment:
A cleaner way to disable these is to set to `null!`

`public IPEndPoint RemoteAddress { get; protected set; } = null!` 

Its cleaner imo.

Reply:
What I really want is to essentially declare all these classes as abstract and require the derived classes' constructors to implement all these. Additionally, I only want the derived classes to have to implement the constructor and not redefine the types (like you have to do for interfaces, right?).

Reply:
Theres no way to do that in c#. If it's defined in the base class, it has to be fully initialized in the base class code only. That analysis doesn't go up to derived classes. And if you make the property abstract, you'd have to define it in every derived class.